---
- name: Verify tests
  hosts: all
  connection: local
  gather_facts: false

  vars:
    device:
      ip_address: "{{ ip_address }}"
      username: "{{ username | default(omit) }}"
      password: "{{ password | default(omit) }}"
      api_key: "{{ api_key | default(omit) }}"

  tasks:
    - name: Security_rule_fail_1
      paloaltonetworks.panos.panos_security_rule:
        provider: '{{ provider }}'
        rule_name: 'Test rule'
        source_zone: ['any']
        destination_zone: ['any']
        category: ['any']
        application: ['ssl']
        service: ['any'] # Service is any
        action: 'allow'
        description: 'Good description goes here'
        log_setting: 'default'

    - name: Security_rule_fail_2
      paloaltonetworks.panos.panos_security_rule:
        provider: '{{ provider }}'
        rule_name: 'Test rule'
        source_zone: ['any']
        destination_zone: ['any']
        category: ['any']
        application: ['ssl']
        service: ['Any'] # Service is any, with uppercase
        action: 'allow'
        description: 'Good description goes here'
        log_setting: 'default'

    - name: Security_rule_fail_3
      paloaltonetworks.panos.panos_security_rule:
        provider: '{{ provider }}'
        rule_name: 'Test rule'
        source_zone: ['any']
        destination_zone: ['any']
        category: ['any']
        application: ['ssl']
        service: [''] # Service is empty
        action: 'allow'
        description: 'Good description goes here'
        log_setting: 'default'

    - name: Security_rule_fail_4
      paloaltonetworks.panos.panos_security_rule:
        provider: '{{ provider }}'
        rule_name: 'Test rule'
        source_zone: ['any']
        destination_zone: ['any']
        category: ['any']
        application: ['ssl']
        action: 'allow'
        description: 'Good description goes here'
        log_setting: 'default'
        # No service provided, defaults to 'any' which is a fail
